modules/plottinggl/glsl/legend.frag

```diff
@@ -60,11 +66,30 @@ vec4 over(vec4 colorB, vec4 colorA) {
     return vec4(col, alpha);
 }
 
+vec4 solidBackground(vec4 src) {
+    // pre-multiply source color and background color
+    src.rgb *= src.a;
+    vec4 bgColor = backgroundColor;
+    bgColor.rgb *= bgColor.a;
+
+    return src + (1.0 - src.a) * bgColor;
+}
+
+vec4 checkerboardBackground(vec4 src, vec2 pos) {
+    return over(checkerBoard(pos), src);
+}
+
+vec4 noBackground(vec4 src) {
+    // use regular OpenGL alpha blending
+    src.rgb *= src.a;
+    return src;    
+}
+
 void main() {
     vec2 texCoord = gl_FragCoord.xy - viewport.xy;
     vec2 outputDim = viewport.zw - vec2(2 * borderWidth);
     vec2 centeredPos = (texCoord - viewport.zw * 0.5);
-    vec2 normPos = texCoord / outputDim;
+    vec2 normPos = (texCoord - vec2(borderWidth)) / outputDim;
```

Comment:
In case the intention is to sample pixel-centered:
(texCoord + 0.5 - borderWidth) / outputDim

Reply:
The reason here is to have normalized coordinates [0,1] within the border corresponding to the given pixel size of the legend. Pixel-centered sampling would be a bonus but requires outputDim-1 for normalization, doesn't it.